adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -71,6 +79,25 @@ public void deleteAllDatasources() {
     StreamSupport.stream(allDatasourceConfigs.spliterator(), true).forEach(amqpPublisher::publishDeletion);
   }
 
+  public void validate(Datasource datasource, DataImport dataImport) throws IOException, ValidationException {   
```

Comment:
Can we return something like 
```javascript
{
    status: SUCCESS / WARNING / ERROR
    messages: ['Error1', 'Error2', ....]
}
```

I think that would make the handling in the `executeImport` method easier.